.github/workflows/check-code-style.yml

```diff
@@ -0,0 +1,134 @@
+# This workflow is intended to check if PR conforms with coding standards used
+# in the project.
+#
+# Documentation for GitHub Actions:
+# [workflow-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+# [context-and-expression-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions
+# [workflow-commands]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions
+
+name: Check code style
+
+on:
+  pull_request:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check formatting for documentation and tests
+      - 'docs/**'
+      - 'test/**'
+
+env:
+  # We need compile command database in order to perform clang-tidy check. So,
+  # in order to perform configure step we need to setup llvm-dev package. This
+  # env variable used to specify desired version of it
+  LLVM_VERSION: 12
+
+jobs:
+  clang-format-and-tidy:
+    name: Run clang-format & clang-tidy
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout sources
+      uses: actions/checkout@v2
+      with:
+        # In order to gather diff from PR we need to fetch not only the latest
+        # commit. Depth of 2 is enough, because GitHub Actions supply us with
+        # merge commit as {{ github.sha }}, i.e. the second commit is a merge
+        # base between target branch and PR
+        fetch-depth: 2
+
+    - name: Gather list of changes
+      id: gather-list-of-changes
+      run: |
+        # GitHub Actions marks step as failed even if one single command has
+        # returned non-zero exit code. This might happen, for example, it we
+        # pipeline git diff --name-only and subsequent grep for files with
+        # particular extensions: if user didn't modify any .cpp files, grep
+        # won't find any matches and return non-zero exit code.
+        # TODO: consider offloading files extension filter to clang-format-diff
```

Comment:
Good point, why don't we leave the filtering and running work in custom scripts?  That will make it easier to run those scripts locally too.

Reply:
Done in 1620536. There are two files (`OpenCL.std.h` and `spirv.hpp`), which are kind of external to us, so we shouldn't expect for them to be compliant with our coding standards (at least this is my understanding). So, I've disabled `clang-format` for them using in-place comments, but `clang-tidy` will still be launched to check changes in there - I think it shouldn't be a blocker, but still a notable change